swabbie-core/src/test/kotlin/com/netflix/spinnaker/swabbie/ResourceTypeHandlerTest.kt

```diff
@@ -652,84 +653,76 @@ object ResourceTypeHandlerTest {
+
+    whenever(
+      notifier.notify(any(), any(), any())
+    ) doReturn NotificationResult(markedResource.resourceOwner, EMAIL, success = true)
+
+    defaultHandler.setCandidates(mutableListOf(defaultResource))
+    defaultHandler.setRules(defaultRules)
+
     defaultHandler.notify(configuration)
-    verify(notifier, times(1)).notify(any(), any(), any())
-    verify(resourceRepository, times(1)).upsert(updatedMarkedResource)
+    expectThat(notificationQueue.isEmpty()).isFalse()
   }
 
   @Test
-  fun `should not send notification if not required`() {
+  fun `should skip scheduling notifications`() {
+    val now = clock.millis()
     val configuration = WorkConfigurationTestHelper.generateWorkConfiguration(
-      notificationConfiguration = NotificationConfiguration(enabled = true, required = false)
-    )
-
-    whenever(dynamicConfigService.getConfig(any(), any(), eq(configuration.maxItemsProcessedPerCycle))) doReturn
-      configuration.maxItemsProcessedPerCycle
-    whenever(ownerResolver.resolve(defaultResource)) doReturn "lucious-mayweather@netflix.com"
-    defaultHandler.setCandidates(mutableListOf(defaultResource))
-    defaultHandler.setRules(defaultRules)
-    whenever(resourceRepository.getMarkedResources()) doReturn listOf(
-      MarkedResource(
-        resource = defaultResource,
-        summaries = listOf(Summary("invalid resource random", "rule 5")),
-        namespace = configuration.namespace,
-        resourceOwner = "test@netflix.com",
-        projectedDeletionStamp = clock.millis().plus(TimeUnit.DAYS.toMillis(3)),
-        markTs = clock.millis()
-      )
+      notificationConfiguration = NotificationConfiguration(enabled = false)
     )
 
-    val updatedMarkedResource = MarkedResource(
+    val markedResource = MarkedResource(
       resource = defaultResource,
       summaries = listOf(Summary("invalid resource random", "rule 5")),
       namespace = configuration.namespace,
-      resourceOwner = "test@netflix.com",
-      projectedDeletionStamp = clock.millis().plus(TimeUnit.DAYS.toMillis(3)),
-      notificationInfo = NotificationInfo(
-        recipient = "",
-        notificationType = NONE.name,
-        notificationStamp = clock.millis()
-      ),
-      markTs = clock.millis()
+      projectedDeletionStamp = now.plus(TimeUnit.DAYS.toMillis(3)),
+      markTs = now
     )
 
+    whenever(
+      dynamicConfigService.getConfig(any(), any(), eq(configuration.maxItemsProcessedPerCycle))
+    ) doReturn configuration.maxItemsProcessedPerCycle
+
+    whenever(
+      resourceRepository.getMarkedResources()
+    ) doReturn listOf(markedResource)
+
+    defaultHandler.setCandidates(mutableListOf(defaultResource))
+    defaultHandler.setRules(defaultRules)
+
     defaultHandler.notify(configuration)
-    verify(notifier, times(0)).notify(any(), any(), any())
-    verify(resourceRepository, times(1)).upsert(updatedMarkedResource)
+
+    with(markedResource.notificationInfo) {
```

Comment:
You can use an expect block in strikt, which i think will make this better. I don't want to have to assert that `this` is not null, reads kind of weird in the test.

You might just be able to do:
```kotlin
val notificationInfo = markedResource.notificationInfo
expect {
  that(notificationInfo).isNotNull()
  ...
  that(notificationInfo.NotificationStamp).isNull()
}
```